src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs

```
@@ -554,17 +554,16 @@ public override void Close()
 
         public override string GetAttribute(int index)
         {
-            // https://github.com/dotnet/runtime/issues/44287
-            //     We should replace returning null with ArgumentOutOfRangeException
-            //     In case of not interactive state likely we should throw InvalidOperationException
             if (!IsInteractive)
             {
-                return null!;
+                throw new InvalidOperationException(SR.InvalidOperation_ExpectedInteractive);
             }
+
             if (index < 0)
             {
-                return null!;
+                throw new ArgumentOutOfRangeException(nameof(index));
             }
+
```

I've missed this before but shouldn't last line of this method also throw ArgumentOutOfRange instead of returning null? The last line is supposedly our of range on the upper side

---

Couldn't the `index <0 ` check extended to a range check of AttributeCount? I.e. `if(index < 0 || index >= AttributeCount)`? However since AttributeCount includes the same logic as here, this would run the same iteration code twice in the worst case.
Otherwise yes, the last line should throw as well.

---

Updated and moved the failing tests to actual test cases as well.